### PR TITLE
Use `ensure_can_withdraw` in chain bridge transfers

### PR DESCRIPTION
**Pull request by branan**
Previously, the chain bridge transfer checked that the total transfer
amount + fee was greater than the "free" balance in the account. This
does not take into account various types of locks that can be on an
account.

This PR changes to use the `ensure_can_withdraw` function from the
Currency trait. This function checks all locks and fails the
transaction if there is not enough available balance in the account.

### CODE REVIEW

**Comment by branan** [comment]
Regarding making this transactional: It looks like while there is a `transactional` macro available, it transforms the function in a way that breaks that `?` operator - I don't think I have enough time this evening to track this down fully, but I'll try. Worst case, this PR does at least fix the known breakage as-is.

**Comment by vedhavyas** [comment]
> It looks like while there is a transactional macro available, it transforms the function in a way that breaks that ? operator 

How much work do you anticipate to get this working ?

**Review by vedhavyas — APPROVED** [review]

File: runtime/src/bridge.rs [hunk 1]
```diff
@@ -257,13 +213,75 @@ impl<T: Trait> Module<T> {
 			.or_else(ensure_root)?;
 		Ok(())
 	}
+
+        fn do_transfer_native(origin: T::Origin, amount: BalanceOf<T>, recipient: Vec<u8>, dest_id: chainbridge::ChainId) -> DispatchResult {
+            let source = ensure_signed(origin)?;
+
+            let token_fee: T::Balance = TokenTransferFee::<T>::get();
+            let currency_token_fee = TryInto::<u128>::try_into(token_fee).map_err(|_| "Token fee not convertible to currency")?
+                                                     .try_into().map_err(|_| "Token fee not convertible to currency")?;
```
Comment (mikiquantum) [inline runtime/src/bridge.rs:222]: Is this `try_into` duplicate?

**Review by mikiquantum — APPROVED** [review]
Minor comment only, very cool that you could get the `transactional` macro in!

File: runtime/src/bridge.rs (hunk 1 above)
Comment (branan) [reply runtime/src/bridge.rs:222]: Alas not redundant. We first convert the T::Balance into a u128, and then from that into a Currency::Balance. I could not figure out how to get the compiler to believe that those two balances were the same type, so we have this unfortunate dance.

**Pull request merged**